src/main/java/frc/team2412/robot/commands/shooter/ShooterTargetCommand.java

```diff
@@ -42,8 +48,30 @@ public void execute() {
             shooter.setHoodAngle(shooterData.getAngle());
             shooter.setFlywheelRPM(shooterData.getRPM());
         }
-        turretAngle = turretEnable.getAsBoolean() ? turretAngle + localizer.getYaw() : 0;
-        shooter.setTurretAngle(turretAngle + localizer.yawAdjustment());
+        if (!turretEnable.getAsBoolean())
+            state = TurretState.STOPPED;
+        else if (turretAngle < ShooterConstants.LEFT_WRAP_THRESHOLD)
+            state = TurretState.WRAP_LEFT;
+        else if (turretAngle > ShooterConstants.RIGHT_WRAP_THRESHOLD)
+            state = TurretState.WRAP_RIGHT;
+        else if (turretAngle > ShooterConstants.LEFT_WRAP && turretAngle < ShooterConstants.RIGHT_WRAP)
+            state = TurretState.TRACKING;
+        switch (state) {
+            case STOPPED:
+                turretAngle = 0;
+                break;
+            case WRAP_LEFT:
+                turretAngle = ShooterConstants.RIGHT_WRAP;
+                break;
+            case WRAP_RIGHT:
+                turretAngle = ShooterConstants.LEFT_WRAP;
+                break;
+            case TRACKING:
+                turretAngle += localizer.getYaw();
+
+        }
+        // shooter.updateTurretAngle(l);
+        shooter.setTurretAngle(turretAngle - (state == TurretState.TRACKING ? localizer.yawAdjustment() : 0));
```

Comment:
Note that we're not checking if the yaw adjustment moves the turret angle past the thresholds.